### ISSUE TITLE
markdown table ignores empty cells

### ISSUE THREAD

**Issue by capitalaslash** [sintaxi/harp#638]
If I try to serve the following markdown file I get a pipe in the second column instead of an empty cell

```markdown
a | b | c
- | - | -
1 |   | 2
```

**Comment by misterhtmlcss** [comment]
I've attached what the code you provided does in an alternative editor. Is my version what you wanted?
![screen shot 2018-10-03 at 5 26 37 pm](https://user-images.githubusercontent.com/13112566/46444984-c65b5780-c731-11e8-9291-543d38e90189.png)

**Comment by misterhtmlcss** [comment]
Please confirm what the issue is, so that I can categorize it and determine how we can assist you.
Thank you for raising your issue!

**Comment by capitalaslash** [comment]
yes, the picture you posted is the expected behavior.
let me know of any further detail you may need.

**Comment by misterhtmlcss** [comment]
@capitalaslash great. 

Also if you can tell me about your OS, Browser, Harp CLI version and anything else you think might be relevant as that's just the basics.

**Comment by capitalaslash** [comment]
OS: CentOS
browser: chromium
version:  0.28.1

I don't know what you mean with html pre-compiler? I am serving a super simple markdown website with a bunch a pages for local-use only with

```
NODE_ENV=production harp server <path> --port <port>
```

**Comment by misterhtmlcss** [comment]
@capitalaslash no worries. If you don't know odds are good you don't use those tools. What I was referring to was .ejs, .pug or .jade file types. These are HTML (template) based languages that compile into regular HTML dynamically.

I'll look into your issue tonight. Thank you for the update! Have a great day.

**Comment by capitalaslash** [comment]
confirmed in 0.29.0

**Comment by misterhtmlcss** [comment]
Too bad. Was hoping for a simple fix. This is getting very clear to me and I'll definitely have a good look tonight. Thank you for jumping through those hoops.

**Comment by misterhtmlcss** [comment]
@capitalaslash I think I have an answer, but it'll disappoint. Markdown doesn't support tables: [Markdown](https://daringfireball.net/projects/markdown/) and we also don't support them. This is a Github thing and we don't support Githubs Markdown. Sorry. I'm new to the project so I had to look it up before I came back to you.

**Comment by misterhtmlcss** [comment]
@capitalaslash I can't see how the issue is bubbling up. I see it as a non-standard .md using tables. Broadly it's not a supported feature. I'll have to speak with one of the team leads to see if it was our intention to support it. I'll let you know.

**Comment by misterhtmlcss** [comment]
@sintaxi thoughts on this? Did we have an intention to support tables with .md?

**Comment by sintaxi** [comment]
@misterhtmlcss That could be as simple as changing the configuration in terraform or switching to a different markdown processor. If we are confident marked is the best processor perhaps we can make the config part of the harp public API. It would make for an easy win.

https://github.com/sintaxi/terraform/blob/master/lib/template/processors/md.js#L4

here are the marked configuration options...

https://marked.js.org/#/USING_ADVANCED.md#options

**Comment by capitalaslash** [comment]
@misterhtmlcss I understand that tables are not "standard" markdown (if you refer to the original implementation), but the table rendering is supported by your back-end, just in a buggy way. You should have it fixed (I hope!), or completely removed (to adhere to that "standard" markdown).
I would like also to point out that there is a work-around for this bug, as you can use an unbreakable space in the column to obtain the expected behavior. Definitely not elegant or intuitive.

**Comment by misterhtmlcss** [comment]
@capitalaslash a PR was made by myself for this issue. @sintaxi is integrating some other bits and pieces as well and then we'll get it up on NPM for you to use. Talk soon!

**Comment by misterhtmlcss** [comment]
@sintaxi to me it looks like the 'marked'-down people are still struggling with issues related to scenarios like this and while I think if we upgrade our build to marked 0.5.x we may overcome this particular issue.

**Comment by capitalaslash** [comment]
I can confirm that marked 0.5 fixes this problem with markedjs/marked#1262

**Comment by misterhtmlcss** [comment]
We'll consider upgrading the Markedjs version to .5.x for inclusion into our 1.x launch. Thank you for confirming my understanding. Always nice to have two heads agree over one.
